src/pipecat/audio/turn/local_smart_turn.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright (c) 2024–2025, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+
+import os
+from typing import Dict
+
+import numpy as np
+import torch
+from loguru import logger
+
+from pipecat.audio.turn.base_smart_turn import BaseSmartTurn
+
+try:
+    import coremltools as ct
+    from transformers import AutoFeatureExtractor
+except ModuleNotFoundError as e:
+    logger.error(f"Exception: {e}")
+    logger.error(
+        "In order to use the LocalSmartTurnAnalyzer, you need to `pip install pipecat-ai[local-smart-turn]`."
+    )
+    raise Exception(f"Missing module: {e}")
+
+
+class LocalCoreMLSmartTurnAnalyzer(BaseSmartTurn):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # To use this locally, set the environment variable LOCAL_SMART_TURN_MODEL_PATH
+        # to the path where the smart-turn repo is cloned.
+        #
+        # Example setup:
+        #
+        #   # Git LFS (Large File Storage)
+        #   brew install git-lfs
+        #   # Hugging Face uses LFS to store large model files, including .mlpackage
+        #   git lfs install
+        #   # Clone the repo with the smart_turn_classifier.mlpackage
+        #   git clone https://huggingface.co/pipecat-ai/smart-turn
+        #
+        # Then set the env variable:
+        #   export LOCAL_SMART_TURN_MODEL_PATH=./smart-turn
+        # or add it to your .env file
+        smart_turn_model_path = os.getenv("LOCAL_SMART_TURN_MODEL_PATH")
```

Comment:
I'd pass a `model_path` argument instead of defining an environment variable. You can use the environment variable on the test files.

Reply:
done.